### PR TITLE
Adding add_free_product_to_new-customer_first_order template

### PR DESCRIPTION
**Pull request by amandahom**
#### PR Review Checklist

Instructions
- [x] Open the Configuration and set the "productvariantid" line to the Variant ID of the Product you would like to add to the order. You can look at the product.json and find the product variant id there. 

Readme
- [ ] Are the install steps clear
- [ ] Do all of the links work

mesa.json
- [x] Key: does it include `shoppad/mesa-templates`, are the subfolders correct?
- [x] Name: is it logical, proper-cased?
- [x] Description: is it logical, end in period?
- [ ] Tags: Do they exist on getmesa.com/templates, proper-cased, logical?
- [x] Source: lowercase
- [x] Destination: lowercase
- [x] Enabled: If no configuration necessary `true`, if there are ANY setup steps `false`
- [x] Do the Input/Output names make sense? How about the keys?
- [ ] Are Storage, Secrets and scripts well-named

Template code
- [x] Is code readable and well-commented?

QA
- [ ] Does the template work

#### Deploy checklist
- [ ] Squash and merge PR
- [ ] Add template key to https://homeroom.theshoppad.com/admin/#/mesa-templates

### CODE REVIEW

File: shopify/order/add_free_product_to_new-customer_first_order/mesa.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
+{
+  "key": "add_free_product_to_new-customer_first_order",
```
Comment (anhdungpham) [inline shopify/order/add_free_product_to_new-customer_first_order/mesa.json:2]: 3. In the key, it looks like you missed changing a dash - to an underscore. Please change to "add_free_product_to_new_customer_first_order".

**Review by anhdungpham — COMMENTED** [review]
@amandahom I wasn't able to get this workflow working. :(

Some things I noticed:

1. I was getting stuck at the Filter step. My `orders_count` starts at 0 when checking the payload. It worked when I changed the Filter to `0`.

Here is the related task: https://www.theshoppad.com/homeroom.theshoppad.com/admin/backdoor/shoppad-ahn-2/mesa/queue/task/6064d9c27c2e6e60555f88f7

2. The Add Line Item to Shopify Order step is failing for me.

Here is the related task: https://www.theshoppad.com/homeroom.theshoppad.com/admin/backdoor/shoppad-ahn-2/mesa/queue/task/6064db50e7f3a91fc52ac764

**Comment by amandahom** [comment]
@anhdungpham Thank you for catching the key typo from the asana task. 

Testing this workflow is so hard on dev stores because of those two issues you pointed out. Could you try testing this workflow on Matt's store? For some reason, the Add line item step doesnt work on my store as well (with the same error) but only on Matt's. 

I think this workflow is okay since its running successfully on this store.. with some fails.. but lets ignore that: https://www.theshoppad.com/homeroom.theshoppad.com/admin/backdoor/moonshot-snacks/mesa/automations/60415409fe0318259f7e5cb5

**Comment by anhdungpham** [comment]
@amandahom I tested on Matt's store and it looks like I am getting the same errors. I wasn't able to get it working.

Here is the link to the workflow on Matt's store: https://www.theshoppad.com/homeroom.theshoppad.com/admin/backdoor/shoppad-matt/mesa/automations/606e1faf1043c809df0a8709/builder

Here are images of the errors: 
Screen Shot 2021-04-07 at 2 17 00 PM

Screen Shot 2021-04-07 at 2 23 35 PM

**Comment by amandahom** [comment]
@anhdungpham Something about dev stores cause these issues. Do you think we should publish this template because Noah has one going too: https://github.com/shoppad/mesa-templates/pull/208/files

It worked when i tested his template. Maybe we could try on this store that worked:
Collab: https://partners.shopify.com/111585/stores/24329912416

Sorry for the back and forth!

**Comment by anhdungpham** [comment]
@amandahom No worries about the back and forth!

I think it would be a nice to have different variations of this template, so that merchants won't have to make the small adjustments. I would still publish this template.

I can test it on that store next week.

**Comment by anhdungpham** [comment]
@amandahom I tried to create a test order on the amanda-shoppad-dev store but I am unable to create a test order.

I see this message at checkout: 

Screen Shot 2021-05-05 at 1 37 51 PM

Is there a way to create test orders?

**Comment by amandahom** [comment]
@anhdungpham Sorry about that. Could you create a draft order and convert it to an order. You can mark the order as "Paid" to convert it.

**Comment by anhdungpham** [comment]
@amandahom Thanks! That worked! QA'd and looks good :)

**Pull request merged**